sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala

```diff
@@ -195,16 +193,16 @@ private[columnar] case object RunLengthEncoding extends CompressionScheme {
       val actualSize = columnType.actualSize(row, ordinal)
       _uncompressedSize += actualSize
 
-      if (lastValue.isNullAt(0)) {
-        columnType.copyField(row, ordinal, lastValue, 0)
+      if (lastValue == null) {
+        lastValue = columnType.clone(columnType.getField(row, ordinal))
```

Comment:
Do we need `clone()` here?  While I guess that this is for `UTF8String`, `lastValue` is used only for keeping a value. Thus, it is ok with just keeping the reference for `UTF8String`.

Reply:
@kiszk Yes it is because `UTF8String`. We reuse `InternalRow` during each iteration and since `UTF8String` is backed by the same memory region in the `InternalRow`, it will be updated each time as we load a new row, which is not correct.

Reply:
got it, thank you for your clarification.